clippy_lints/src/methods/waker_clone_and_wake.rs

```diff
@@ -0,0 +1,34 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet_with_applicability;
+use clippy_utils::{match_def_path, paths};
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_span::sym;
+
+use super::WAKER_CLONE_AND_WAKE;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>, recv: &'tcx Expr<'_>) {
+    let ty = cx.typeck_results().expr_ty(recv).peel_refs();
+
+    if let Some(did) = ty.ty_adt_def() &&
+        match_def_path(cx, did.did(), &paths::WAKER) &&
+        let ExprKind::MethodCall(func, waker_ref, &[], _) = recv.kind &&
+        func.ident.name == sym::clone
+    {
+        let mut applicability = Applicability::MachineApplicable;
+
+        span_lint_and_sugg(
+            cx,
+            WAKER_CLONE_AND_WAKE,
+            expr.span,
```

Comment:
I did that because some other lints already do.

For example,
```rust
pub fn test(seeker: &mut dyn io::Seek) -> io::Result<u64> {
    seeker.seek(io::SeekFrom::Current(0))
}
```

results in
```
warning: using `SeekFrom::Current` to start from current position
  --> src/main.rs:18:5
   |
   |     seeker.seek(io::SeekFrom::Current(0))
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace with: `seeker.stream_position()`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#seek_from_current
   = note: `#[warn(clippy::seek_from_current)]` on by default
```

But both are probably used, so just tell me.